dftimewolf/lib/exporters/gce_disk_export.py

```diff
@@ -36,34 +42,42 @@ def __init__(self, state, critical=False):
           the entire recipe to fail if the module encounters an error.
     """
     super(GoogleCloudDiskExport, self).__init__(state, critical=critical)
-    self.analysis_project = None
     self.source_project = None
-    self.source_disk = None
     self.gcs_output_location = None
-    self.exported_disk_name = None
-    self._image_name = None
+    self.analysis_project = None
+    self.remote_instance_name = None
+    self.disk_names = []
+    self.all_disks = False
+    self.source_disks = []
+    self.exported_image_name = None
 
   def Process(self):
     """Creates and exports disk image to the output bucket."""
-    image_object = self.analysis_project.compute.CreateImageFromDisk(
-        self.source_disk, name=self._image_name)
-    image_object.ExportImage(
-        self.gcs_output_location, output_name=self.exported_disk_name)
-    image_object.Delete()
-    output_url = os.path.join(
-        self.gcs_output_location, '{0:s}.tar.gz'.format(
-            self.exported_disk_name))
-    self.logger.info('Disk was exported to: {0:s}'.format(output_url))
-    container = containers.URL(path=output_url)
-    self.state.StoreContainer(container)
+    for source_disk in self.source_disks:
+      image_object = self.analysis_project.compute.CreateImageFromDisk(
+          source_disk)
+      # If self.exported_image_name = None, default output_name is
+      # {src_disk.name}-{TIMESTAMP('%Y%m%d%H%M%S')}.tar.gz
+      image_object.ExportImage(
+          self.gcs_output_location, output_name=self.exported_image_name)
+      image_name = self.exported_image_name or image_object.name
+      image_object.Delete()
+      output_url = os.path.join(
+          self.gcs_output_location,
+          '{0:s}.tar.gz'.format(image_name))
+      self.logger.info('Disk was exported to: {0:s}'.format(output_url))
+      container = containers.URL(path=output_url)
+      self.state.StoreContainer(container)
 
   # pylint: disable=arguments-differ
   def SetUp(self,
             source_project_name,
-            source_disk_name,
             gcs_output_location,
             analysis_project_name=None,
-            exported_disk_name=None):
+            disk_names=None,
```

Comment:
rename this to `source_disk_names`, so as to be consistent with the previous parameter name.

Reply:
Done